tests/functional/lifecycle/LifecycleTask.js

```diff
@@ -0,0 +1,770 @@
+            newParams.MaxKeys = this.overrideMax;
+        }
+
+        return {
+            send: this._listObjectsHandler.bind(this, newParams),
+            on: this._noop,
+            httpRequest: { headers: {} },
+        };
+    }
+
+    _listObjectsHandler(params, cb) {
+        const bucket = this.s3Storage[params.Bucket];
+        if (!bucket) {
+            return cb(errors.NoSuchBucket);
+        }
+
+        const bucketObjs = bucket.Objects;
+        if (!bucketObjs || bucketObjs.length === 0) {
+            const res = { Contents: [] };
+            return cb(null, res);
+        }
+
+        let markerIdx = 0;
+        let maxIdx = params.MaxKeys;
+        if (params.Marker) {
+            markerIdx = bucketObjs.findIndex(o => o.Marker === params.Marker);
+            maxIdx = markerIdx + params.MaxKeys;
+        }
+
+        const IsTruncated = bucketObjs.length > maxIdx;
+        const Contents = bucketObjs.slice(markerIdx, maxIdx).map(o => {
+            const newObj = Object.assign({}, o);
+            delete newObj.TagSet;
+            return newObj;
+        });
+        const res = {
+            Contents,
+            IsTruncated,
+            NextMarker: IsTruncated ? this._findNextMarker(bucketObjs,
+                Contents[Contents.length - 1].Marker, 'Marker') : null,
+        };
+        return cb(null, res);
+    }
+
+    /**
+     * get all versions from bucket
+     * @param {Object} params - parameters
+     * @param {string} params.Bucket - bucket name
+     * @param {string} [params.KeyMarker] - optional key marker
+     * @param {string} [params.VersionIdMarker] - required if key marker exists
+     * @return {Object} wrapper to mock `send`
+     */
+    listObjectVersions(params) {
+        assert(params.Bucket);
+        if (params.VersionIdMarker || params.KeyMarker) {
+            assert(params.VersionIdMarker && params.KeyMarker);
+        }
+
+        // Override default MaxKeys from LifecycleTask
+        const newParams = params;
+        if (this.overrideMax) {
+            newParams.MaxKeys = this.overrideMax;
+        }
+
+        return {
+            send: this._listObjectVersionsHandler.bind(this, newParams),
+            on: this._noop,
+            httpRequest: { headers: {} },
+        };
+    }
+
+    _listObjectVersionsHandler(params, cb) {
+        const bucket = this.s3Storage[params.Bucket];
+        if (!bucket) {
+            return cb(errors.NoSuchBucket);
+        }
+
+        const bucketVersions = bucket.Versions;
+        if (!bucketVersions || bucketVersions.length === 0) {
+            const res = { Versions: [] };
+            return cb(null, res);
+        }
+
+        let markerIdx = 0;
+        let maxIdx = params.MaxKeys;
+        if (params.KeyMarker) {
+            markerIdx = bucketVersions.findIndex(v =>
+                v.KeyMarker === params.KeyMarker &&
+                v.VersionId === params.VersionIdMarker);
+            if (markerIdx === -1) {
```

Comment:
Just to note, these errors here were for myself to validate that I correctly implemented the mock functions. I can remove these if requested 👍